apps/route_patterns/lib/repo.ex

```diff
@@ -43,7 +44,10 @@ defmodule RoutePatterns.Repo do
   end
 
   def by_stop_id(stop_id) do
-    [stop: stop_id]
+    routes = RoutesRepo.by_stop(stop_id)
+    route_ids = Enum.join(Enum.map(routes, fn r -> r.id end), ",")
+
+    [route: route_ids]
```

Comment:
**issue**: I am noticing a lot more headsigns shown, and I am thinking that this change might be reintroducing the problem we had where we're showing headsigns that aren't applicable to this stop.

can you try re-adding the stop argument here? e.g. use `[route: route_ids, stop: stop_id]` for the API request, and we can see if that helps.

there are likely many examples but if you'd like a reference I am using stop `1260`,  in this PR having three headsigns, but on dev it shows Route 22 to **Ashmont** only:
<img width="600" alt="image" src="https://github.com/mbta/dotcom/assets/2136286/1881a186-3833-4cf6-b54e-f63f70049f8c">